ftw/keywordwidget/widget.py

```diff
@@ -221,7 +221,7 @@ def get_choice_field(self):
 
     def show_add_term_field(self):
         if isinstance(self.choice_field, ChoicePlus):
-            return self.field.value_type.allow_new
+            return getattr(api.portal.getRequest(), 'allow_new', True)
```

Comment:
This approach could be a problem when using multiple widgest in the same request, as we are not sure for which instance the `allow_new` counts. This can be a problem when using multiple keyword widgets in the same schame and configuring different add permissions.

After a short grep I think that the `ChoiceField` does not need to know the `allow_new` value: it is just a very strange implementation. The only place I could find where the `allow_new` attribute is read is exactly this line.

So I suggest to do these changes:
- Remove all instances of `allow_new` from the `ChoiceField`
- remove the permission-check from the update-method completely
change the return statement here to:
```python
return api.user.has_permission(self.add_permission, obj=self.context)
```

Am I missing something?